### ISSUE TITLE
Optionally use IAM authentication

### ISSUE THREAD

**Issue by mcanevet** [prometheus/mysqld_exporter#749]
It would be great to be able to use IAM authentication when the exporter runs on AWS: https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.Connecting.Go.html

**Comment by tejavegesna-lb** [comment]
Agree with @mcanevet We need a way to use IAM Authentication for metrics

**Comment by lasith-kg** [comment]
Agree! Support for dynamic credentials (of the likes of RDS IAM Auth) would be great from a security posture
